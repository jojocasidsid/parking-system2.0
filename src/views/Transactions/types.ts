export interface ITransactionProperties {
	id: number
	vehicle: string
	price: number
	transactionDate: string
	parkingType: number
	parkTime: string
	hours: number
}

export interface IListTransactions<Entity> {
	_order: 'asc' | 'desc' | undefined
	_sort: keyof Entity | ''
	_page: number
	_limit: number
	q: string
}

export interface ITransactionListColums {
	id: keyof ITransactionProperties | ''
	label: string
	type?: 'primary'
	sortable: boolean
}
