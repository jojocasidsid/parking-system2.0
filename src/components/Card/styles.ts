import styled from '@emotion/styled'

export const CardWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-direction: column;
	border-radius: 30px;
	border: 2px solid ${({ theme }) => theme.palette.secondary.main};
`
export const StyledSlotHeader = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`
