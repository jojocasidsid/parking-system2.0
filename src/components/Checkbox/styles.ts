import styled from '@emotion/styled'

export const StyledRoot = styled('div')`
	padding: 18px;
`
