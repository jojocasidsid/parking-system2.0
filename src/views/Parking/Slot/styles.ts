import styled from '@emotion/styled'
import DirectionsCarIcon from '@mui/icons-material/DirectionsCar'

export const StyledSlot = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`

export const StyledImage = styled(DirectionsCarIcon)`
	font-size: 45px;
`

export const StyledBody = styled('div')`
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	text-align: center;
`
