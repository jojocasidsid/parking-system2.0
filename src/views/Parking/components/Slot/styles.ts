import styled from '@emotion/styled'
import DirectionsCarIcon from '@mui/icons-material/DirectionsCar'

export const ThumbnailWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-direction: column;
	border-radius: 30px;
	border: 2px solid ${({ theme }) => theme.palette.secondary.main};
`
export const StyledSlot = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`

export const StyledImage = styled(DirectionsCarIcon)`
	font-size: 45px;
`

export const ActionWrapper = styled('div')`
	& > button {
		margin: 20px;
	}
`
