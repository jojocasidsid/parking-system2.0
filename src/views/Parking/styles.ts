import styled from '@emotion/styled'

export const StyledRoot = styled('div')`
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	padding: 20px;
`

export const ParkingWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-wrap: nowrap;
`

export const SlotsWrapper = styled('div')`
	display: grid;
	grid-template-columns: repeat(5, 1fr);
	grid-template-rows: repeat(5, 1fr);
	grid-gap: 10px;
	width: 90vw;
`
