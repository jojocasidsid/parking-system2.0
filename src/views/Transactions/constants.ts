import { ITransactionListColums } from './types'

export const COLUMNS: ITransactionListColums[] = [
	{
		id: 'id',
		label: 'ID',
		sortable: true,
	},
	{
		id: 'vehicle',
		label: 'Plate Number',
		sortable: false,
	},
	{
		id: 'parkingType',
		label: 'Parking Type',
		sortable: true,
	},

	{
		id: 'hours',
		label: 'Hours Spent',
		sortable: true,
	},

	{
		id: 'price',
		label: 'Price',
		sortable: false,
	},

	{
		id: 'transactionDate',
		label: 'Transaction Date',
		sortable: true,
	},
]
