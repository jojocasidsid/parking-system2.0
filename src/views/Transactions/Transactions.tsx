import React from 'react'
// packages
import { Paper } from '@mui/material'

// queries
import { useFetchEarnings } from 'queries'

// hooks
import useListingTable from 'hooks/useListingTable'

// components
import ListingTable from 'components/ListingTable'

// constants
import { COLUMNS } from './constants'

const Transactions = () => {
	const { filters, setFilters, debouncedFilters } = useListingTable({
		_order: 'asc',
		_sort: '',
		_page: 1,
		_limit: 10,
		q: '',
	})

	const { isLoading, data: EarningsData } = useFetchEarnings(
		debouncedFilters,
		filters
	)

	return (
		<div>
			<Paper>
				<ListingTable
					title='Transaction Listings'
					rows={EarningsData?.data || []}
					columns={COLUMNS}
					isFilters
					filters={filters}
					setFilters={setFilters}
					isPagination
					totalPages={Number(EarningsData?.headers['x-total-count']) || 1}
					isLoading={isLoading}
				/>
			</Paper>
		</div>
	)
}

export default Transactions
