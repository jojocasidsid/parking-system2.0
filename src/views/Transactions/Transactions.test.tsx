import React from 'react'
import renderer from 'react-test-renderer'

import { QueryClient, QueryClientProvider } from 'react-query'
import { SnackbarProvider } from 'notistack'
import { ThemeProvider } from '@mui/material/styles'

import theme from 'config/theme'

import Transactions from './Transactions'

const queryClient = new QueryClient()

describe('Transactions Page', () => {
	it('should render Transaction Page without crashing', () => {
		const tree = renderer.create(
			<QueryClientProvider client={queryClient}>
				<ThemeProvider theme={theme}>
					<SnackbarProvider
						maxSnack={5}
						anchorOrigin={{
							vertical: 'bottom',
							horizontal: 'right',
						}}
						preventDuplicate
						autoHideDuration={5000}
					>
						<Transactions />
					</SnackbarProvider>
				</ThemeProvider>
			</QueryClientProvider>
		)

		expect(tree).toMatchSnapshot()
	})
})
