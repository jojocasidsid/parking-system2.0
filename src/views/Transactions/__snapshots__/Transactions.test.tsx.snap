// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions Page should render Transaction Page without crashing 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-14rutos-MuiPaper-root"
  >
    <div
      aria-labelledby="tableHeading"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-1x0o62d-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-1hptjqf-MuiGrid-root"
          >
            <h1
              className="MuiTypography-root MuiTypography-h1 css-1m29lhz-MuiTypography-root"
              id="tableHeading"
            >
              Transaction Listings
            </h1>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item css-1boge50-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
            >
              <div
                className="css-evn1e9"
              >
                <label
                  className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-uj5iz7-MuiFormLabel-root-MuiInputLabel-root"
                  htmlFor="search"
                >
                  Search
                </label>
                <div
                  className="MuiFormControl-root MuiTextField-root css-1yfph09-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1k534au-MuiInputBase-root-MuiOutlinedInput-root"
                    disableUnderline={true}
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled={false}
                      id="searchBar"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Search Transactions"
                      required={false}
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-hdw1oc"
                      >
                        <span
                          className="notranslate"
                        >
                          â€‹
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiBox-root css-8atqhb"
        >
          <hr
            className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          />
        </div>
        <div
          className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
        >
          <table
            aria-labelledby="tableTitle"
            className="MuiTable-root MuiTable-stickyHeader css-vlh9os-MuiTable-root"
            role={null}
          >
            <thead
              className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head css-1fztqu1-MuiTableRow-root"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-13wn5r-MuiTableCell-root"
                  scope="col"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <span
                    className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
                    >
                      ID
                    </p>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m7 10 5 5 5-5z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-13wn5r-MuiTableCell-root"
                  scope="col"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <p
                    className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
                  >
                    Plate Number
                  </p>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-13wn5r-MuiTableCell-root"
                  scope="col"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <span
                    className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
                    >
                      Parking Type
                    </p>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m7 10 5 5 5-5z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-13wn5r-MuiTableCell-root"
                  scope="col"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <span
                    className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
                    >
                      Hours Spent
                    </p>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m7 10 5 5 5-5z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-13wn5r-MuiTableCell-root"
                  scope="col"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <p
                    className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
                  >
                    Price
                  </p>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-13wn5r-MuiTableCell-root"
                  scope="col"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <span
                    className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
                    >
                      Parking Time
                    </p>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m7 10 5 5 5-5z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-13wn5r-MuiTableCell-root"
                  scope="col"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <span
                    className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
                    >
                      Leave Time
                    </p>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m7 10 5 5 5-5z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-13wn5r-MuiTableCell-root"
                  scope="col"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <span
                    className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
                    >
                      Transaction Date
                    </p>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m7 10 5 5 5-5z"
                      />
                    </svg>
                  </span>
                </th>
              </tr>
            </thead>
          </table>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container css-1ry2eeo-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="MuiBox-root css-gmuwbf"
              data-testid="loading-indicator"
            >
              <span
                className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-19r7jhb-MuiCircularProgress-root"
                role="progressbar"
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-czpcox-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item css-10k7qhr-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-grid-xs-auto css-14on6sx-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-62c4vu-MuiTypography-root"
                >
                  Show
                </p>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  className="css-dwoubo"
                >
                  <div
                    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary css-1kjct6a-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                    onClick={[Function]}
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="_limit"
                      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-jcfq8n-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                      id="_limit"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      10
                    </div>
                    <input
                      aria-hidden={true}
                      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      name="_limit"
                      onAnimationStart={[Function]}
                      onChange={[Function]}
                      tabIndex={-1}
                      value={10}
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-hdw1oc"
                      >
                        <span
                          className="notranslate"
                        >
                          â€‹
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-62c4vu-MuiTypography-root"
                >
                  transactions
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
            />
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1tuue3l-MuiGrid-root"
            >
              <nav
                aria-label="pagination navigation"
                className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
              >
                <ul
                  className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
                >
                  <li>
                    <button
                      aria-label="Go to previous page"
                      className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-1cjqjaa-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      type="button"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateBeforeIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to next page"
                      className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext css-1cjqjaa-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateNextIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
